ompi/mca/coll/ftagree/coll_ftagree_earlyreturning.c

```diff
@@ -3234,7 +3235,11 @@ int mca_coll_ftagree_era_inter(void *contrib,
     ompi_group_free(&uniongrp);
     shadowcomm->c_contextid = comm->c_contextid;
     shadowcomm->c_epoch = comm->c_epoch;
-    snprintf(shadowcomm->c_name, MPI_MAX_OBJECT_NAME, "SHADOW OF %s", &comm->c_name[0]);
+    // Use opal_snprintf() so that we don't get a compiler warning
```

Comment:
We should ditch the comment.  We have opal_snprintf, which means we should always use it instead of snprintf...

Reply:
Actually, digging deeper, I see in the description of `opal_snprintf` in at https://github.com/open-mpi/ompi/blob/651be3974d3029cfd387d87070f62ea6072c08a9/opal/util/printf.h#L52

We *do* actually have a `#define snprintf opal_snprintf` in `opal_config_bottom.h`, but only in the case where the compiler does not provide `opal_snprintf` https://github.com/open-mpi/ompi/blob/651be3974d3029cfd387d87070f62ea6072c08a9/opal/include/opal_config_bottom.h#L381-L383

It looks like `opal_config_bottom.h` has been this way since Open MPI v1.1.x (i.e., pretty much the very beginning).

If we think we should be using `opal_snprintf()` everywhere, should we make this `#define` be unconditional (i.e., _always_ define `snprintf` to be `opal_snprintf`)?

I only see 3 other places in the OMPI code base where we actually use `opal_snprintf`, and they all have Jeff-like comments saying "We're using opal_snprintf() to avoid a stupid compiler warning."

Reply:
Ah, fair enough.  We're really inconsistent on when the opal_<function> calls are mandatory or not.  I actually dislike them, but we do mandate them in some situations (like opal_asprintf).  I really dislike this work around for the compiler warning; I wonder if we wouldn't be better off just disabling that warning?